docs/contributing/pullrequest-submission-and-lifecycle.md

```diff
@@ -70,7 +70,42 @@ based on these guidelines to speed up the review and merge process.
 
 ### Go Coding Style
 
-The following Go language resources provide common coding preferences that may be referenced during review, if not automatically handled by the project's linting tools.
+All Go code is automatically checked for compliance with various linters, such as `gofmt`. These tools can be installed using the `GNUMakefile` in this repository.
+
+```console
+% cd terraform-provider-aws
+% make tools
+```
+
+Check your code with the linters:
+
+```console
+% make lint
+```
+
+`gofmt` will also fix many simple formatting issues for you:
+
+```console
+% gofmt -w -s ./aws
+```
+
+The import statement in a Go file follows these rules (see [#15903](https://github.com/hashicorp/terraform-provider-aws/issues/15903)):
+
+1. Import declarations are grouped into a maximum of three groups:
+    - Standard packages (also called short import path or built-in packages)
+    - Third-party packages (also called long import path packages)
+    - Local packages
+2. Groups are in that order (i.e., standard packages, third-party packages, and local packages)
+3. Groups are separated by a single blank line
+4. Packages within each group are alphabetized
+
+Check your imports:
+
+```console
+% make importlint
+```
```

Comment:
Not for this pull request, but we should also have a target to apply automatic fixes for imports.

Reply:
I don't know of anything existing that fixes all the issues. `goimports` fixes some and `impi` identifies many but I ended up manually/regex fixing many issues.